qt/python/mantidqt/widgets/sliceviewer/peaksviewer/test/test_peaksviewer_peaksviewercollectionpresenter.py

```diff
@@ -82,6 +85,99 @@ def test_notify_called_for_each_subpresenter(self, mock_create_model):
+                                                         presenter.clear_handle, presenter.delete_handle)
+
+        CreateSampleWorkspace(OutputWorkspace="ws")
+        CreateSampleWorkspace(OutputWorkspace="ws")
+
+        presenter.replace_handle.assert_called_once()
+
+    def test_ensure_that_the_ads_observer_calls_rename_handle(self, _):
+        presenter = PeaksViewerCollectionPresenter(MagicMock())
+        presenter.rename_handle = MagicMock()
+        self.assertTrue(isinstance(presenter._ads_observer, SliceViewerADSObserver))
+        presenter._ads_observer = SliceViewerADSObserver(presenter.replace_handle, presenter.rename_handle,
+                                                         presenter.clear_handle, presenter.delete_handle)
+
+        CreateSampleWorkspace(OutputWorkspace="ws")
+        RenameWorkspace(InputWorkspace="ws", OutputWorkspace="ws1")
+
+        presenter.rename_handle.assert_called_once_with("ws", "ws1")
+
+    def test_ensure_replace_handle_removes_and_re_adds_workspaces(self, _):
+        presenter = PeaksViewerCollectionPresenter(MagicMock())
+        presenter.append_peaksworkspace = MagicMock()
+        presenter.workspace_names = MagicMock(return_value=["ws"])
+        presenter.remove_peaksworkspace = MagicMock()
+
+        presenter.replace_handle("ws", None)
+
+        presenter.remove_peaksworkspace.assert_called_once_with("ws")
+        presenter.append_peaksworkspace.assert_called_once_with("ws")
+
+    def test_ensure_delete_handle_removes_workspace(self, _):
+        presenter = PeaksViewerCollectionPresenter(MagicMock())
+        presenter.remove_peaksworkspace = MagicMock()
+        presenter.workspace_names = MagicMock(return_value=["ws"])
+
+        presenter.delete_handle("ws")
+
+        presenter.remove_peaksworkspace.assert_called_once_with("ws")
+
+    def test_ensure_clear_handle_removes_all_workspaces(self, _):
+        presenter = PeaksViewerCollectionPresenter(MagicMock())
+        presenter.remove_peaksworkspace = MagicMock()
+        presenter.workspace_names = MagicMock(return_value=["ws", "ws1", "ws2"])
+
+        presenter.clear_handle()
+
+        presenter.remove_peaksworkspace.assert_any_call("ws")
+        presenter.remove_peaksworkspace.assert_any_call("ws1")
+        presenter.remove_peaksworkspace.assert_any_call("ws2")
+        self.assertEqual(3, presenter.remove_peaksworkspace.call_count)
+
+    def test_ensure_rename_handle_removes_and_re_adds_the_new_workspace_name(self, _):
+        presenter = PeaksViewerCollectionPresenter(MagicMock())
+        presenter.remove_peaksworkspace = MagicMock()
+        presenter.workspace_names = MagicMock(return_value=["ws"])
+        presenter.append_peaksworkspace = MagicMock()
+
+        presenter.rename_handle("ws", "ws1")
+
+        presenter.remove_peaksworkspace.assert_called_once_with("ws")
+        presenter.append_peaksworkspace.assert_called_once_with("ws1")
+
+    def test_ensure_close_called
```

Comment:
Did you intend to add another test here, or it it covered by the sliceviewer presenter tests?

Reply:
AH I must have not finished that test! I will go do that now! This is why we have code review!